Makefile

```diff
@@ -0,0 +1,103 @@
+# Makefile
+
+WARNINGS=RUSTDOCFLAGS="-D warnings"
+DEBUG_ASSERTIONS=RUSTFLAGS="-C debug-assertions"
+
+# -- linting --------------------------------------------------------------------------------------
+
+# Runs Clippy with configs
+.PHONY: clippy
+clippy:
+	cargo clippy --workspace --all-targets --all-features -- -D warnings
```

Comment:
Should we not run `clippy` against nightly as well?

Reply:
I believe that we had removed nightly in a previous PR because it was making the CI fail in some cases in our previous impl. 

I believe that we could add it back now.